lexicon/providers/corenetworks.py

```diff
@@ -0,0 +1,280 @@
+            data = {}
+        if query_params is None:
+            query_params = {}
+
+        self._log( "url: %s with data %s and query_params %s" % ( url, str(data), str(query_params) ) )
+        default_headers = {}
+
+        if self.token:
+            default_headers['Authorization'] = "Bearer {0}".format(self.token)
+
+        response = requests.request(action,
+            self.api_endpoint + url,
+            params=query_params,
+            data=json.dumps(data),
+            headers=default_headers,
+        )
+        # if the request fails for any reason, throw an error.
+        response.raise_for_status()
+        if response.text and response.json() is None:
+            raise Exception('No data returned')
+
+        return response.json() if response.text else None
+
+    # Ref: https://beta.api.core-networks.de/doc/#functon_dnszones
+    def _list_zones(self):
+        return self._get('/dnszones/')
+
+    def _make_identifier(self, rtype, name, content):
+        return hashlib.sha1('/'.join([ rtype, name, content ]).encode('utf-8')).hexdigest()
+
+    def _refresh_auth_file(self):
+        """Retrieve token and zones from json file"""
+        try:
+            auth = open(self.auth_file_path, "r")
+            if auth.mode == "r":
+                self.auth_file = json.loads(auth.read())
+                auth.close()
+                return True
+            else:
+                auth.close()
+                return False
+        except IOError as e:
+            LOGGER.debug("No stored authentication found: %s. Acquiring token via API call." % os.strerror(e.errno))
+            self._get_token()
+            return True
+
+    def _commit_auth_file(self):
+        """Store authentication into json file."""
+        try:
+            auth = open(self.auth_file_path, "w")
+            if auth.mode == "w":
+                content = json.dumps(self.auth_file)
+                auth.write(content)
+                os.chmod(self.auth_file_path, 0o600)
+                return True
+            else:
+                return False
+        except IOError as e:
+            LOGGER.debug("Could not write authentication file: %s" % os.strerror(e.errno))
+        finally:
+            auth.close()
+
+    def _get_token(self):
+        """Request new token via API call"""
+        LOGGER.debug("Entering _get_token.")
+        data = {
+            'login'   : self._get_provider_option('auth_username'),
+            'password': self._get_provider_option('auth_password')
+        }
+        self._log(str(data))
+        payload = self._post('/auth/token', data = data)
+        LOGGER.debug("%s", str(payload))
+        self.token  = payload['token']
+        self.expiry = payload['expires'] + time.time()
+
+        # Prepare auth file and commit changes
+        self.auth_file['token']  = self.token
+        self.auth_file['expiry'] = self.expiry
+        self._commit_auth_file()
+
+    def _log(self, message):
```

Comment:
This is probably a leftover from the development? Contains sensitive data.

Reply:
Yes, you are right. I will fix it with the next commit.